### PR TITLE
nixos/tools: continue cleanup

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

- remove with lib
- shuffle some things around for readability
- add separate options for each tool
- 
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/installer/tools/tools.nix [hunk 1]
```diff
@@ -104,145 +213,48 @@ in
...
-        nixos-install
-        nixos-rebuild
-        nixos-generate-config
-        nixos-option
-        nixos-version
-        nixos-enter
-      ];
+  imports = let
+    mkToolModule = { name, package ? pkgs.${name} }: { config, ... }: {
+      options.system.tools.${name}.enable = lib.mkEnableOption "${name} script" // {
```
Comment (Mic92) [inline nixos/modules/installer/tools/tools.nix:218]: Do you have a specific use-case for making all these tools toggle-able?

File: pkgs/tools/nix/nixos-install-tools/default.nix [hunk 2]
```diff
@@ -19,6 +20,8 @@ in
 (buildEnv {
   name = "nixos-install-tools-${version}";
   paths = lib.attrValues {
+    inherit nixos-enter;
+
```
Comment (Mic92) [inline pkgs/tools/nix/nixos-install-tools/default.nix:24]:
```suggestion
```

File: nixos/modules/installer/tools/tools.nix (hunk 1 above)
Comment (K900) [reply nixos/modules/installer/tools/tools.nix:218]: I mostly want my server boxes to not have anything but nixos-version, but it's easier to make all of them optional than to bikeshed over which exact ones should be.

**Comment by Mic92** [comment]
I think I messed up the resolving the merge conflict. It's probably just easier to do a local rebase instead.

**Comment by K900** [comment]
Rebased.

File: nixos/modules/installer/tools/tools.nix [hunk 3]
```diff
@@ -44,23 +35,141 @@ let
...
+
+    { config, lib, pkgs, ... }:
+
+    {
+      imports =
+        [ # Include the results of the hardware scan.
+          ./hardware-configuration.nix
+        ];
+
+    $bootLoaderConfig
```
Comment (Mic92) [inline nixos/modules/installer/tools/tools.nix:62]: We were debating the other day that this section is kind of hardware related. But this another pull request.

Comment (K900) [reply nixos/modules/installer/tools/tools.nix:62]: I still kinda want to rewrite the whole thing tbh

**Comment by Mic92** [comment]
@ofborg test installer.simple

File: nixos/modules/installer/tools/tools.nix (hunk 3 above)
Comment (Mic92) [reply nixos/modules/installer/tools/tools.nix:62]: What do you want to rewrite this to? We are also working on a replacement as well.

**Pull request merged**